projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -179,6 +179,47 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    /**
+     * Navigates to the specific child grid based on the array of paths leading to it
+     *
+     * @param pathToChildGrid Array of IPathSegments that describe the path to the child grid
+     * each segment is described by the rowKey of the parent row and the rowIslandKey.
+     */
+    public navigateToChildGrid(pathToChildGrid: IPathSegment[], cb?: () => void) {
+        if (pathToChildGrid.length == 0) {
+            if (cb) {
+                cb();
+            }
+            return;
+        }
+        const pathElem = pathToChildGrid.shift();
+        const rowKey = pathElem.rowKey;
+        const rowIndex = this.grid.gridAPI.get_rec_index_by_id(rowKey, this.grid.dataView);
+        // scroll to row, since it can be out of view
+        this.performVerticalScrollToCell(rowIndex, -1, () => {
+            this.grid.cdr.detectChanges();
+            // next, expand row, if it is collapsed
+            const row = this.grid.getRowByIndex(rowIndex);
+            if (!row.expanded) {
+                row.expanded = true;
+                // update sizes after expand
+                this.grid.verticalScrollContainer.recalcUpdateSizes();
+                this.grid.cdr.detectChanges();
+            }
+
+            const childGrid =  this.grid.gridAPI.getChildGrid([pathElem]);
+            const positionInfo = this.getElementPosition(childGrid.nativeElement, false);
+            if (positionInfo.offset > 0) {
+                this.grid.verticalScrollContainer.addScrollTop(positionInfo.offset);
+                this.grid.verticalScrollContainer.chunkLoad.pipe(first()).subscribe(() => {
+                    childGrid.navigation.navigateToChildGrid(pathToChildGrid, cb);
+                });
+            } else {
+                childGrid.navigation.navigateToChildGrid(pathToChildGrid, cb);
```

Comment:
Is there a case where you can hit that because you are comparing the position of the childGrid nativeElement with the parent positioning? IMO, that `positionInfo.offset` will be > 0 always.

Reply:
I guess not, since it always scrolls to the parent row first.